packages/app/src/app/overmind/effects/deployment/githubPages.ts

```diff
@@ -0,0 +1,111 @@
+import getTemplate from '@codesandbox/common/lib/templates';
+import { Sandbox } from '@codesandbox/common/lib/types';
+import axios from 'axios';
+
+const GHPagesBaseUrl = 'https://builder.csbops.io/gh-pages';
+
+type Options = {
+  provideJwtToken: () => Promise<string | null>;
+};
+
+export default (() => {
+  let _options: Options;
+  let _jwtToken: string | undefined;
+  const createHeaders = (jwt: string) =>
+    jwt
+      ? {
+          Authorization: `Bearer ${jwt}`,
+        }
+      : {};
+
+  return {
+    initialize(options: Options) {
+      _options = options;
+    },
+    async provideJwtCached() {
+      if (!_jwtToken) {
+        try {
+          const token = await _options.provideJwtToken();
+          setTimeout(() => {
+            // Token expires after 10 seconds, for safety we actually cache the token
+            // for 5 seconds
+            _jwtToken = undefined;
+          }, 5000);
+          return token;
+        } catch (e) {
+          _jwtToken = undefined;
+          return Promise.reject(e);
+        }
+      }
+
+      return _jwtToken;
+    },
+
+    async getSite(id: string) {
+      const url = `${GHPagesBaseUrl}/${id}`;
+      const token = await this.provideJwtCached();
+
+      const { data } = await axios.get(url, {
+        headers: createHeaders(token),
+      });
+      return data;
+    },
+
+    async getLogs(id: string) {
+      const url = `${GHPagesBaseUrl}/${id}/status`;
+      const token = await this.provideJwtCached();
+
+      const { data } = await axios.get(url, {
+        headers: createHeaders(token),
+      });
+
+      return data;
+    },
+
+    async deploy(sandbox: Sandbox, username: string): Promise<any> {
+      const token = await this.provideJwtCached();
+      const template = getTemplate(sandbox.template);
+      const buildCommand = (name: string) => {
+        if (name === 'styleguidist') {
+          return 'styleguide:build';
+        }
+        if (name === 'nuxt') {
+          return 'generate';
+        }
+
+        if (name === 'parcel') {
+          return `build --public-url /csb-${sandbox.id}/`;
+        }
+
+        if (name === 'preact-cli') {
+          return 'build --no-prerender';
+        }
+
+        if (name === 'gatsby') {
+          return 'build --prefix-paths';
+        }
+
+        return 'build';
+      };
+
+      const env = (name: string) => {
+        if (name === 'create-react-app') {
+          return `PUBLIC_URL=https://${username}.github.io/csb-${sandbox.id}/`;
+        }
+        return '';
+      };
+
+      const website = await axios.post(
```

Comment:
Not related to this PR, but important to evaluate anyway. @andrasbacsai is the builder isolated from everything? These are commands that can break out of the system, essentially remote code execution as a service (I know we're good at this :) ), but we need to be extra mindful that people will be able to run anything. And we need to make sure that people can't break out and read files in if private sandboxes are deployed with this service.

Reply:
We do send the JWT so you can only access your own sandboxes and build those, also sent this to andras for answering

Reply:
Good point. I'll check this. Details will be in private.

Reply:
What @SaraVieira said, it's true. From that point of view, everybody could build their own sandboxes, nothing more. Our API does authorization.

Reply:
I see, so if I have a build script that runs `ls` and sends the output to one of my servers, I won't be able to see files of other sandboxes that are building at the same time?

Reply:
Nop, it starts its own service for each sandbox